tests/core/gomock/BUILD.bazel

```diff
@@ -0,0 +1,33 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library", "go_test", "gomock")
```

Comment:
Nit: this shouldn't be in `tests/core` since it isn't a part of the core `go` package, it is a part of `extras`. Should we put this inside `extras/tests/gomock`?

Reply:
I am trying to be consistent with go_embed_data, which is also in extras. I can create a follow up PR to move both `tests/core/go_embed_data` and `tests/core/gomock` to `tests/extras`.